src/content/en/fundamentals/engage-and-retain/push-notifications/sending-messages-with-web-push-libraries.md

```diff
@@ -0,0 +1,324 @@
+
+When it comes to sending a push message we ultimately need some event to trigger the process of
+ sending a message to users. A common approach would be creating an admin page that let's you
+ configure and trigger the push message. But you could create a program to run locally or any
+ other approach that allows accessing the list of PushSubscriptions and running the code to
+ trigger the push message.
+
+Our demo has an "admin like" page that lets you trigger a push. Since it's just a demo it's a
+ public page.
+
+I'm going to go through each step involved in getting the demo working, these will be baby
+ steps to everyone follow along, including anyone who is new to node.
+
+When we discussed subscribing a user we covered adding an `applicationServerKey` to the
+ `subscribe()` options. It's on the back end that we'll need the private key.
+
+> Remember you can use the web-push tool to generate application server keys or use
+ web-push-codelab.appspot.com to generate some application server keys. See ["How to Create
+ Application Server
+ Keys"](http://localhost:4000/chapter-02/01-subscribing-a-user/#how-to-create-application-server-keys)
+ for more details.
+
+In the demo these values are added to our node app like so (boring code I know, but just want
+ you to know there is no magic):
+
+
+    const vapidKeys = {
+      publicKey:
+ 'BEl62iUYgUivxIkv69yViEuiBIa-Ib9-SkvMeAtA3LFgDzkrxZJjSgSnfckjBJuBkr3qBUYIHBQFLXYp5Nksh8U',
+      privateKey: 'UUxI4O8-FbRouAevSmBQ6o18hgE4nSG3qwvJTfKc-ls'
+    };
+
+
+Next we need to install the `web-push` module for our node server:
+
+    npm install web-push --save
+
+Then in our node script we require in the `web-push` module
+like so:
+
+
+    const webpush = require('web-push');
+
+
+Now we can start to use the `web-push` module. First we need to tell the web-push module about
+ our application server keys (remember they are also known as VAPID keys because that's the
+ name of the spec).
+
+
+    const vapidKeys = {
+      publicKey:
+ 'BEl62iUYgUivxIkv69yViEuiBIa-Ib9-SkvMeAtA3LFgDzkrxZJjSgSnfckjBJuBkr3qBUYIHBQFLXYp5Nksh8U',
+      privateKey: 'UUxI4O8-FbRouAevSmBQ6o18hgE4nSG3qwvJTfKc-ls'
+    };
+    
+    webpush.setVapidDetails(
+      'mailto:web-push-book@gauntface.com',
+      vapidKeys.publicKey,
+      vapidKeys.privateKey
+    );
+
+
+We also include a "mailto:" string as well. This string needs to be either a URL or a mailto
+ email address. This piece of information will actually be sent to web push service as part of
+ the request to trigger a push. The reason this is done is so that if a web push service needs
+ to get in touch, they have some information that will enable them to (i.e. a URL, hopefully
+ with some contact information or an email address).
+
+With this, the `web-push` module is ready to use, the next step is to trigger a push message.
+
+The demo uses the pretend admin panel to trigger push messages.
+
+![Screenshot of the Admin Page.](./images/demo-admin-page.png)
```

Comment:
I tried but it messes up the code examples :(

Can just remove it if you would prefer.